### DISCUSSION TITLE
WebRtc video not displayed in safari

### DISCUSSION THREAD

**Discussion by FGLoytec** [sipsorcery-org/sipsorcery#1258] [Q&A]
Hi,

I have implement a WebRtc cameras server (using sipsorcery of course, thanks to the example): IP camera -> ffmpeg -> WebRtc -> browser

The video codec used is H264.

I test it on a MAC mini using safari browser (latest version)

The Rtc peer connection is correctly connected but the video is not display.

looking for the event on the video control, I got this: loadstart progress suspend

On the MAC, I have also a chrome browser installed and trying the same, it works fine, I got this: loadstart progress loadmetadata loadeddata canplay canplaythrough playing

the video control on the html page is declared like this: <video controls autoplay playsinline muted id="videoCtl" style="max-width:100%;height:auto;" oncontextmenu="return false;" disablepictureinpicture></video>

Does anyone have an idea why the video stay suspend or have a clue or tips to make it work?

Thanks in advance

**Comment by sipsorcery** [comment]
Does the [MP4](https://github.com/sipsorcery-org/sipsorcery/blob/master/examples/WebRTCExamples/WebRTCMp4Source/Program.cs) example work? It's a good way to test wether FFMpeg and the video ecndoign is working without involving a webcam.

**Comment by FGLoytec** [comment]
Hi, thanks for the quick reply.

I will try it.

BR

**Comment by FGLoytec** [comment]
Hi,

I made the test and yes, the MP4 example works fine.

So it seems to be an encoding problem.

Looking at the example, the encoding set by the example is VP8.

I will try to use VP8 instead of H264 to see if it works.

BR
